lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs

```diff
@@ -471,6 +569,30 @@ impl FilesystemBackend {
         }
     }
 
+    /// Called from the Go code when a [`tdp::SharedDirectoryListResponse`] is received from the browser.
+    ///
+    /// Calls the [`SharedDirectoryListResponseHandler`] associated with the completion id of the
+    /// [`tdp::SharedDirectoryListResponse`].
+    pub fn handle_tdp_sd_list_response(
+        &mut self,
+        tdp_resp: tdp::SharedDirectoryListResponse,
+    ) -> PduResult<()> {
+        if let Some(handler) = self
```

Comment:
`match` here might be nicer

Reply:
```rust
    /// Called from the Go code when a [`tdp::SharedDirectoryListResponse`] is received from the browser.
    ///
    /// Calls the [`SharedDirectoryListResponseHandler`] associated with the completion id of the
    /// [`tdp::SharedDirectoryListResponse`].
    pub fn handle_tdp_sd_list_response(
        &mut self,
        tdp_resp: tdp::SharedDirectoryListResponse,
    ) -> PduResult<()> {
        match self
            .pending_sd_list_resp_handlers
            .remove(&tdp_resp.completion_id)
        {
            Some(handler) => handler.call(self, tdp_resp),
            None => Err(custom_err!(
                "FilesystemBackend::handle_tdp_sd_list_response",
                FilesystemBackendError(format!(
                    "received invalid completion id: {}",
                    tdp_resp.completion_id
                ))
            )),
        }
    }
```

Agreed this is a bit easier to read. This pattern is used repeatedly throughout the stack, so I'll add a PR at the tip that refactors all of them.